src/index.js

```diff
@@ -1,39 +1,151 @@
+  taskList.innerHTML = '';
+
+  tasks.forEach((task, index) => {
+    const taskElement = document.createElement('li');
+    taskElement.className = 'task';
+    taskElement.draggable = true;
+    taskElement.dataset.index = task.index;
+
+    const taskCheckbox = document.createElement('input');
+    taskCheckbox.type = 'checkbox';
+    taskCheckbox.checked = task.completed;
+    taskCheckbox.addEventListener('change', (event) => {
+      tasks[index].completed = event.target.checked;
+      saveTasks(tasks);
+      renderTaskList();
+    });
+
+    const taskLabel = document.createElement('label');
+    taskLabel.textContent = task.description;
+
+    const taskinput = document.createElement('input');
+    taskinput.type = 'text';
+    taskinput.className = 'disable';
+    taskinput.id = 'editfield';
+
+    const taskDeleteButton = document.createElement('button');
+    taskDeleteButton.type = 'button';
+    taskDeleteButton.innerHTML = '<i class="fa-solid fa-ellipsis-vertical"></i>';
+    taskDeleteButton.addEventListener('click', () => {
+      tasks.splice(index, 1);
+      updateIndex();
+      saveTasks(tasks);
+      renderTaskList();
+    });
+    const taskeditButton = document.createElement('button');
+    taskeditButton.type = 'button';
+    taskeditButton.innerHTML = '<i class="fa fa-pencil"></i>';
+    const tasksaveButton = document.createElement('button');
+    tasksaveButton.type = 'button';
+    tasksaveButton.innerHTML = '<i class="fa fa-floppy-o"></i>';
+    tasksaveButton.className = 'disable';
+
+    taskeditButton.addEventListener('click', () => {
+      tasksaveButton.classList.remove('disable');
+      taskinput.classList.remove('disable');
+      taskinput.value = taskLabel.textContent;
+      taskinput.focus();
+      taskLabel.classList.add('disable');
+      taskeditButton.classList.add('disable');
+    });
+
+    tasksaveButton.addEventListener('click', () => {
+      tasksaveButton.classList.add('disable');
+      taskinput.classList.add('disable');
+      taskLabel.textContent = taskinput.value;
+      tasks[index].description = taskinput.value;
+      taskLabel.classList.remove('disable');
+      taskeditButton.classList.remove('disable');
+      saveTasks(tasks);
+      renderTaskList();
+    });
+
+    taskElement.appendChild(taskCheckbox);
+    taskElement.appendChild(taskLabel);
+    taskElement.appendChild(taskinput);
+    taskElement.appendChild(taskeditButton);
+    taskElement.appendChild(tasksaveButton);
+    taskElement.appendChild(taskDeleteButton);
+
+    taskList.appendChild(taskElement);
   });
-}
+};
+
+const addTask = (name) => {
+  const task = { index: tasks.length + 1, description: name, completed: false };
+  tasks.push(task);
+  saveTasks(tasks);
+  renderTaskList();
+};
+
+const clearCompletedTasks = () => {
+  tasks = tasks.filter((task) => !task.completed);
+  saveTasks(tasks);
+  renderTaskList();
+};
+
+const handleDragStart = (event) => {
+  event.target.classList.add('dragging');
+};
```

Comment:
- [Optional] Impressive job @atok624, implementing the drag and drop functionality. Though it's possible to drag and arrange items on your website, your functionality stops working once I rearrange items and throws exceptions. Such cases will cause bad user experiences. I suggest implementing the functionality if possible or removing it till it is completely functional. 🙌 

https://user-images.githubusercontent.com/69077061/221162482-69f12cca-82f9-4739-93a0-6b42d98d5aa4.mp4

Reply:
Okay 
Thank you
I will surely work on that

Reply:
perfect 👌